packages/store/src/store.ts

```diff
@@ -93,6 +101,23 @@ export class Store {
     return selectorFn(this._stateStream.getValue());
   }
 
+  /**
+   * Select a signal from the state.
+   */
+  selectSignal<T>(
+    selector: (state: any, ...states: any[]) => T,
+    options?: CreateComputedOptions<T>
+  ): Signal<T>;
+  selectSignal<T = any>(
+    selector: string | Type<any>,
+    options?: CreateComputedOptions<T>
+  ): Signal<T>;
```

Comment:
We can remove some of the bad "non type-safe" NGXS legacy for this new API. We should therefore not support specifying a string or a function as arguments for this new API. This also means that specifying a state class should not be supported.

We should only support `TypedSelector<TModel>` as the sole argument. Essentially this constrains our users to use the `selectSignal` function with a typed selector or a StateToken, both of which carry type information.